packages/web/vite.config.ts

```diff
@@ -30,14 +28,35 @@ const fixAcceptHeader404 = () => ({
   }
 })
 
-export default defineConfig(({ mode }) => {
-  const env = loadEnv(mode, process.cwd(), 'VITE_')
+export default defineConfig(async ({ mode }) => {
+  // Despite loading env here, the result is the same as a filtered process.env
+  // rather than dynamically loading the correct env file by mode.
+  // Since the build/start scripts use turbo, and other packages don't use vite,
+  // --mode isn't an allowed parameter. Instead, each script sets the
+  // environment manually using env-cmd.
+  // The only exception is test (vitest), which does not set the env manually,
+  // and uses the "test" mode to load .env.test which sets VITE_ENVIRONMENT to
+  // "development". Even in that case, process.env gets explicitly set
+  // anyway because that's what the application code looks at.
+  // Despite loading .env.production (which notably doesn't exist anyway),
+  // loadEnv prioritizes process.env anyway so we're not at risk of overrides.
+  const env = loadEnv(mode, path.join(process.cwd(), 'env'), 'VITE_')
```

Comment:
⚠️ scary, someone check to make sure I got that right!

There might be a world in which we can use `--mode` but if it ain't broke don't fix it 🤷

Reply:
This all looks ok to me!